go-controller/pkg/ovn/policy.go

```diff
@@ -650,7 +650,7 @@ func (oc *Controller) handleLocalPodSelector(
 		return
 	}
 
-	np.podHandlerList = append(np.podHandlerList, h)
+	np.podHandlerMap[np.namespace] = h
```

Comment:
This won't work; each NetworkPolicy in a Namespace needs its own podHandler. Maybe you need `map[string][]*factory.Handler` ?

Reply:
ok, yeah, so I was misremembering how this all works... `np` represents only a single `NetworkPolicy`. However, we add a podHandler for each clause of the NetworkPolicy that selects pods. eg, maybe:

    - from:
      - namespaceSelector:
          matchLabels:
            label1: one
        podSelector:
          matchLabels:
            app: one
      - namespaceSelector:
          matchLabels:
            label2: two
        podSelector:
          matchLabels:
            app: two

If there's some namespace that has both `label1=one` and `label2=two` then it would need two podHandlers. So you'd need to do something like keeping track of which `podHandler` goes with which clause of the policy. (eg, "this one goes with the 3rd `From` element of the 2nd `Ingress` element"...) Yuck.

FWIW we did this totally differently in openshift-sdn; there we just have one pod watcher, and when a pod is added/updated/deleted, we check its labels against any NetworkPolicy that has podSelectors that might match in its namespace.